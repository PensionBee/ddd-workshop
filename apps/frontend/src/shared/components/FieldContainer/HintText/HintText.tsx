import { DisclaimerR } from "@shared/components/Typography";

export type HintTextProps = {
  /** Hint text to be displayed */
  hintText?: string;
};

export const HintText: React.FC<HintTextProps> = ({ hintText }) => {
  if (!hintText) return null;
  return (
    <DisclaimerR color="grey1" mt={8}>
      {hintText}
    </DisclaimerR>
  );
};
