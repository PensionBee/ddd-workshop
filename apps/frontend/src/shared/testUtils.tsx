import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import type { Breakpoints } from "./styling/breakpoints";
import { BREAKPOINTS } from "./styling/breakpoints";
import { theme } from "./styling/theme";
import { vi } from "vitest"

type ReactTestingLibraryProviderProps = {
  children: React.ReactNode;
};

// eslint-disable-next-line react-refresh/only-export-components
const ReactTestingLibraryProviders: React.FC<
  ReactTestingLibraryProviderProps
> = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const customRender = (ui: React.ReactElement, options?: any) => {
  return render(ui, { wrapper: ReactTestingLibraryProviders, ...options });
};

// re-export everything
// eslint-disable-next-line react-refresh/only-export-components
export * from "@testing-library/react";

// override render method
export { customRender as render };

export const sleep = (delay: number) => {
  return new Promise((resolve) => setTimeout(resolve, delay));
};

export const mockMediaMatches = (breakpoint: Breakpoints) => {
  Object.defineProperty(window, "matchMedia", {
    writable: true,
    value: (query: string) => {
      const breakpointValue = BREAKPOINTS[breakpoint];
      const queryValue = parseInt(query.replace(/^\D+/g, ""));
      const breakpointValues = [...Object.values(BREAKPOINTS)];
      if (!breakpointValues.includes(queryValue)) {
        throw new Error(
          `Invalid breakpoint value: ${queryValue}, this likely means there has been a change to the way that breakpoints are setup.`
        );
      }
      return {
        matches: queryValue <= breakpointValue,
        addEventListener: vi.fn(),
        removeEventListener: vi.fn(),
      };
    },
  });
};

export const routerRender = (ui: React.ReactNode, pathname = "/") => {
  return customRender(
    <MemoryRouter initialEntries={[{ pathname }]}>{ui}</MemoryRouter>
  );
};
