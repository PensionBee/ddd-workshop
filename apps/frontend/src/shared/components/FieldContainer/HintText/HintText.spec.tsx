import { render } from "@shared/testUtils";
import { vi } from "vitest";
import { HintText } from "./HintText";

const HINT_TEXT = "Hint text";

describe("FieldWrapper/HintText", () => {
  it("should render correctly", () => {
    const { getByText, asFragment } = render(<HintText hintText={HINT_TEXT} />);

    expect(getByText(HINT_TEXT)).toBeVisible();
    expect(asFragment()).toMatchSnapshot();
  });
  it("should not render if no hintText", () => {
    const { container } = render(<HintText />);

    expect(container.firstChild).toBeNull();
  });
});
