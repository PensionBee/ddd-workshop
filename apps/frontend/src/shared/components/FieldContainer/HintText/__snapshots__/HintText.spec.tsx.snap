// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldWrapper/HintText should render correctly 1`] = `
<DocumentFragment>
  <span
    class="sc-bcXHqe TOXwu"
    data-rc="DisclaimerR"
  >
    Hint text
  </span>
</DocumentFragment>
`;
